lib/stackops/apiGateway.js

```diff
@@ -278,9 +278,9 @@ module.exports = function(currentTemplate, aliasStackTemplates, currentAliasStac
 		// Adjust permission to reference the function aliases
 		_.forOwn(apiLambdaPermissions, (permission, name) => {
 			const functionName = _.replace(name, /LambdaPermissionApiGateway$/, '');
-			const versionName = _.find(_.keys(versions), version => _.startsWith(version, functionName));
-			const aliasName = _.find(_.keys(aliases), alias => _.startsWith(alias, functionName));
-			const isExternalRef = isExternalRefAuthorizerPredicate(permission.Properties.FunctionName); 
+			const versionName = _.find(_.keys(versions), version => _.startsWith(version, `${functionName}LambdaVersion`));
```

Comment:
this will work, but you essentially not checking for `_.startsWith` anymore but for string equality. Shouldn't then a simple `===` suffice?
May be to be more robust this should be _startsWith(version

Reply:
`getFunctionVersionName` and `getAliasVersionName` functions added to utils